browser/ui/browser_commands.cc

```diff
@@ -185,15 +188,37 @@ void CloseWalletBubble(Browser* browser) {
 #endif
 }
 
-void CopyCleanLink(Browser* browser) {
+void CopyCleanLink(Browser* browser, const GURL& url) {
   GURL sanitized_url = brave::URLSanitizerServiceFactory::GetForBrowserContext(
                            browser->profile())
-                           ->SanitizeURL(browser->tab_strip_model()
-                                             ->GetActiveWebContents()
-                                             ->GetVisibleURL());
+                           ->SanitizeURL(url);
 
   ui::ScopedClipboardWriter scw(ui::ClipboardBuffer::kCopyPaste);
   scw.WriteText(base::UTF8ToUTF16(sanitized_url.spec()));
 }
 
+void CopyLinkWithStrictCleaning(Browser* browser, const GURL& url) {
```

Comment:
it feels like we need to apply cleaning only to `http(s)://` links? Probably somewhere should be a quick return for non-interesting links

Reply:
There are no restrictions by http(s). if we want to add it then I think this should be explained in UI as well, @fmarier wdyt?

Reply:
Good catch, we should limit this to http/https URLs. @spylogsster would you mind filing a follow-up issue for this?